src/main/java/org/sonarqube/gradle/AndroidUtils.java

```diff
@@ -0,0 +1,132 @@
+        populateMainPropsForAndroid(properties, bootClassPath, releaseVariant);
+
+        populateTestPropsForAndroid(properties, bootClassPath, releaseVariant);
+      }
+    });
+    project.getPlugins().withType(TestPlugin.class, appPlugin -> {
+      TestExtension androidExtension = project.getExtensions().getByType(TestExtension.class);
+      if (androidExtension != null) {
+        List<File> bootClassPath = androidExtension.getBootClasspath();
+        ApplicationVariant releaseVariant = androidExtension.getApplicationVariants().stream().filter(v -> RELEASE.equals(v.getName())).findFirst().orElse(null);
+        populateMainPropsForAndroid(properties, bootClassPath, releaseVariant);
+
+        populateTestPropsForAndroid(properties, bootClassPath, releaseVariant);
+      }
+    });
+  }
+
+  private static void populateTestPropsForAndroid(Map<String, Object> properties, List<File> bootClassPath, TestedVariant releaseVariant) {
+    UnitTestVariant unitTestVariant = releaseVariant.getUnitTestVariant();
+    if (unitTestVariant != null) {
+      List<File> testDirs = unitTestVariant.getSourceSets().stream().map(AndroidUtils::getFilesFromSourceSet).collect(
+        ArrayList::new,
+        ArrayList::addAll,
+        ArrayList::addAll);
+      properties.put(SonarQubePlugin.SONAR_TESTS_PROP, SonarQubePlugin.nonEmptyOrNull(testDirs.stream().filter(SonarQubePlugin.FILE_EXISTS).collect(Collectors.toList())));
+
+      List<File> testLibraries = new ArrayList<>();
+      testLibraries.addAll(bootClassPath);
+      testLibraries.addAll(unitTestVariant.getJavaCompiler().getClasspath().getFiles());
+      SonarQubePlugin.setTestClasspathProps(properties, unitTestVariant.getJavaCompiler().getDestinationDir(), testLibraries);
+    }
+  }
+
+  @NotNull
+  private static List<File> populateMainPropsForAndroid(Map<String, Object> properties, List<File> bootClassPath, BaseVariant releaseVariant) {
+    List<File> srcDirs = releaseVariant.getSourceSets().stream().map(AndroidUtils::getFilesFromSourceSet).collect(
+      ArrayList::new,
+      ArrayList::addAll,
+      ArrayList::addAll);
+    properties.put(SonarQubePlugin.SONAR_SOURCES_PROP, SonarQubePlugin.nonEmptyOrNull(srcDirs.stream().filter(SonarQubePlugin.FILE_EXISTS).collect(Collectors.toList())));
+
+    properties.put(SonarQubePlugin.SONAR_JAVA_SOURCE_PROP, releaseVariant.getJavaCompiler().getSourceCompatibility());
+    properties.put(SonarQubePlugin.SONAR_JAVA_TARGET_PROP, releaseVariant.getJavaCompiler().getTargetCompatibility());
+
+    List<File> libraries = new ArrayList<>();
+    libraries.addAll(bootClassPath);
+    libraries.addAll(releaseVariant.getJavaCompiler().getClasspath().getFiles());
+    SonarQubePlugin.setMainClasspathProps(properties, false, releaseVariant.getJavaCompiler().getDestinationDir(), libraries);
+    return bootClassPath;
+  }
+
+  private static List<File> getFilesFromSourceSet(SourceProvider sourceSet) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "getFilesFromSourceSet" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUnusedPrivateMethod)